src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -444,6 +443,12 @@
     ),
     TrimParametersGrammar=Nothing(),
     TemporaryGrammar=Nothing(),
+    JoinLikeClauseGrammar=Sequence(
+        AnySetOf(
+            Ref("PivotUnpivotStatementSegment"),
+            min_times=1,
+        ),
+    ),
```

Comment:
Don't think we need `Sequence` and `AnySetOf`?

```suggestion
    JoinLikeClauseGrammar=AnySetOf(
        Ref("PivotUnpivotStatementSegment"),
        min_times=1,
    ),
```